game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -145,14 +145,13 @@ public void setUnits(
       final Collection<Unit> defending,
       final Collection<Unit> attacking,
       final Collection<Unit> bombarding,
-      final Collection<Unit> amphibious,
+      final boolean amphibious,
```

Comment:
WDYT of using an enum here to represent the different battle states? It would be purely for a more descriptive API

Reply:
Such as `BattleType`?  You recently modified it for the different air battles.  Maybe add `AMPHIBIOUS` there as well?

Reply:
That is not a bad idea. I think there are two options:
1. a new enum with `AMPHIBIOUS` and `NON_AMPHIBIOUS`
2. add an entry for battle type

There pros & cons to either approach.

Reply:
I looked into what the `amphibious` was used for and after tracking down its usage, I don't think it is actually needed.  I think that 'amphibious' can be determined by looping through the units and seeing if one of them has a `wasAmphibious`.

I wasn't completely certain because of a few uses of `AbstractBattle#isAmphibious` but I felt that the use here could be removed.  So, I didn't do any enum.